frontend/src/components/Panel.svelte

```diff
@@ -0,0 +1,94 @@
+<script lang="ts">
+
+    import GraphOptions from "./GraphOptions.svelte";
+    import ColorLegend from "./ColorLegend.svelte";
+    import EventDetail from "./EventDetail.svelte";
+    import QueryForm from "./QueryForm.svelte";
+    import type { GraphSettings } from "../layout";
+    import type { FixedQuery } from "../uitypes";
+    import type { FullEvent } from "../apidefinition";
+    
+    //Boolean variables
+    export let show_legend_placeholder: boolean; 
+    export let show_menu_placeholder: boolean; 
+    export let awaiting_query_request: boolean; 
+    export let current_query_changed: boolean; 
+
+    //Object variables; used for onClick actions.
+    export let reset_graph_options_placeholder: any; 
+    export let consume_query: any; 
+    export let use_selected_as_root: any; 
+    export let add_filter: any; 
+    export let submit_state_query: any; 
+
+    export let current_query: FixedQuery; 
+    export let  qhistory: FixedQuery[];  
+    export let selected_node: FullEvent | null;
+    export let graph_options: GraphSettings;
+    export let colors:{}; 
```

Comment:
Preferably stated with object instead of {} for readability.

Reply:
I saw `{}` being used before. But if `object` is a better practice, then sure.

Reply:
> The design after syncing master

The image has not been uploaded (But the image you sent me looked nice :D)

Reply:
Just out of curiosity, what happens when one opens graph options? How is the overflow handled? (I will try it out later today when I find time)

Reply:
> Just out of curiosity, what happens when one opens graph options? How is the overflow handled? (I will try it out later today when I find time)

For the time being the legend is hidden once the option is clicked, as it originally was. There is not enough room for all three panels. Since the filter position is fixed I cannot add scrolling. Once the filter is updated we can work on that.